### ISSUE TITLE
system not turning off switch after set-temperature has been reached

### ISSUE THREAD

**Issue by LittlePapaJohn** [custom-components/climate.programmable_thermostat#24]
The system does not seem to be able to turn off the switch when temperature is reached. The reall temperature does not change in the frontend after that although the sensor still updates. I added the errorr message that I recieve below. I get this error every 30 seconds. Manually turning off switch works and resets all and the error message stops.   I have a rlink switch (kaku) that turns the boiler off via a potential free switch
Turning on works.
 
Info:
Version: 6.3
Hassio: 0.117.1
Switch: Kaku RFlink
Operation in heating - cooling automatic mode

error:
2020-11-01 09:39:15 ERROR (MainThread) [homeassistant] Error doing job: Task exception was never retrieved
Traceback (most recent call last):
  File "/config/custom_components/programmable_thermostat/climate.py", line 305, in _async_sensor_changed
    await self.control_system_mode()
  File "/config/custom_components/programmable_thermostat/climate.py", line 234, in control_system_mode
    await self._async_control_thermo(mode="heat")
  File "/config/custom_components/programmable_thermostat/climate.py", line 352, in _async_control_thermo
    await self._async_turn_off(mode=mode)
  File "/config/custom_components/programmable_thermostat/climate.py", line 257, in _async_turn_off
    related_climate_hvac_action = self.hass.states.get(self._related_climate).attributes['hvac_action']
KeyError: 'hvac_action'

**Comment by MapoDan** [comment]
Ciao,

Have you set related_climate?
It seems that it is checking for the HVAC_ACTION of the related climate without finding it.

Can you paste the config set up of your climate?
Thanks

**Comment by LittlePapaJohn** [comment]
Thanks for the hint. 
I removed the related climate and it is working.

I had my evo-home system as related climate. I have 2 rooms and a home climate entity (average of 2) I tried all three climate entries but all three causes the same issues. I also tried to look for any clue to find out whether it is on or not. this system does not provide feedback on that and perhaps this is causing that issue.
I therefore close this bug request.

Keep up the good work and best regards,
John
